File: neo4j-service/src/test/java/renovation/neo4j/service/web/controller/StartupControllerTest.java
/*
 * Created under not commercial project "Renovation"
 *
 * Copyright 2021-2025
 */

package renovation.neo4j.service.web.controller;

import org.apache.http.HttpStatus;
import org.hamcrest.CoreMatchers;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import renovation.common.web.RestTestInit;
import renovation.neo4j.service.web.Route;

import static renovation.common.util.JsonUtil.simplify;

@Tag("componentTest")
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class StartupControllerTest extends RestTestInit {

    public StartupControllerTest() {
        super(Route.STARTUP);
    }

    @Test
    void about() {
        getRequest().get("/about")
                .then()
                .statusCode(HttpStatus.SC_OK)
                .body(CoreMatchers.equalTo(
                        simplify(
                                """
                                        {
                                            "name"        : "neo4j-service",
                                            "description" : "Neo4j service module"
                                        }
                                        """)
                ));
    }

    @Test
    void index() {
        getRequest().get("/module")
                .then()
                .statusCode(HttpStatus.SC_OK)
                .body(CoreMatchers.equalTo("Hi, it's \"Neo4j-service\" module"));
    }
}

File: kafka-service/src/test/kotlin/renovation/kafka/service/StartupControllerTest.kt
/*
 * Created under not commercial project "Renovation"
 *
 * Copyright 2021-2023
 */

package renovation.kafka.service

import io.restassured.RestAssured
import io.restassured.module.kotlin.extensions.Then
import io.restassured.module.kotlin.extensions.When
import kotlin.test.BeforeTest
import kotlin.test.Test
import org.apache.http.HttpStatus.SC_OK
import org.hamcrest.CoreMatchers
import org.junit.jupiter.api.Tag
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.web.server.LocalServerPort
import renovation.common.util.Json

@Tag("integrationTest")
@SpringBootTest(
    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
)
internal class StartupControllerTest(
    @LocalServerPort val port: Int
) {
    @BeforeTest
    fun init() {
        RestAssured.port = port
    }

    @Test
    fun `Assert project controller`() {
        When {
            get("/module")
        }.Then {
            statusCode(SC_OK)
            body(CoreMatchers.equalTo("Hi, it's \"Renovation/kafka-service\" module"))
        }
    }

    @Test
    fun `Assert about controller`() {
        When {
            get("/about")
        }.Then {
            statusCode(SC_OK)
            body(
                CoreMatchers.equalTo(
                    Json.rowJson(
                        """
                        {
                            "name" : "renovation/kafka-service",
                            "description" : "Kafka-service module of renovation project"
                        }
                        """
                    )
                )
            )
        }
    }
}
